client/lib/posts/test/post-edit-store.js

```diff
@@ -265,38 +265,144 @@ describe( 'post-edit-store', () => {
 		assert( PostEditStore.isDirty() );
 	} );
 
-	test( 'excludes metadata without an operation on edit', () => {
-		const postEdits = {
-				title: 'Super Duper',
-				metadata: [
-					{ key: 'super', value: 'duper', operation: 'update' },
-					{ key: 'foo', value: 'bar', operation: 'delete' },
-					{ key: 'bar', value: 'foo' },
-				],
-			},
-			expectedMetadata = [
-				{ key: 'super', value: 'duper', operation: 'update' },
-				{ key: 'foo', value: 'bar', operation: 'delete' },
-			];
-
+	test( 'updates existing metadata on edit', () => {
+		// initial post
 		dispatcherCallback( {
 			action: {
 				type: 'RECEIVE_POST_TO_EDIT',
-				post: {},
+				post: {
+					metadata: [
+						{ key: 'keepable', value: 'constvalue' },
+						{ key: 'updatable', value: 'oldvalue' },
+						{ key: 'deletable', value: 'trashvalue' },
+					],
+				},
 			},
 		} );
 
+		// apply some edits
+		const postEdits = {
+			title: 'Super Duper',
+			metadata: [
+				{ key: 'updatable', value: 'newvalue', operation: 'update' },
+				{ key: 'deletable', operation: 'delete' },
+			],
+		};
+
 		dispatcherCallback( {
 			action: {
 				type: 'EDIT_POST',
 				post: postEdits,
 			},
 		} );
 
-		assert( PostEditStore.get().metadata === postEdits.metadata );
+		// check the expected values of post attributes after the edit is applied
 		assert( PostEditStore.get().title === postEdits.title );
+		assert(
+			isEqual( PostEditStore.get().metadata, [
+				{ key: 'keepable', value: 'constvalue' },
+				{ key: 'updatable', value: 'newvalue', operation: 'update' },
+				{ key: 'deletable', operation: 'delete' },
+			] )
+		);
+
+		// check the modifications sent to the API endpoint
 		assert( PostEditStore.getChangedAttributes().title === postEdits.title );
-		assert( isEqual( PostEditStore.getChangedAttributes().metadata, expectedMetadata ) );
+		assert(
+			isEqual( PostEditStore.getChangedAttributes().metadata, [
+				{ key: 'updatable', value: 'newvalue', operation: 'update' },
+				{ key: 'deletable', operation: 'delete' },
+			] )
+		);
+	} );
+
+	test( 'should include metadata edits made previously', () => {
+		// initial post
+		dispatcherCallback( {
+			action: {
+				type: 'RECEIVE_POST_TO_EDIT',
+				post: {
+					metadata: [ { key: 'deletable', value: 'trashvalue' } ],
+				},
+			},
+		} );
+
+		// first edit
+		dispatcherCallback( {
+			action: {
+				type: 'EDIT_POST',
+				post: {
+					metadata: [ { key: 'deletable', operation: 'delete' } ],
+				},
+			},
+		} );
+
+		// second edit
+		dispatcherCallback( {
+			action: {
+				type: 'EDIT_POST',
+				post: {
+					metadata: [ { key: 'updatable', value: 'newvalue', operation: 'update' } ],
+				},
+			},
+		} );
+
+		assert(
+			isEqual( PostEditStore.get().metadata, [
+				{ key: 'deletable', operation: 'delete' },
+				{ key: 'updatable', value: 'newvalue', operation: 'update' },
+			] )
+		);
+	} );
+
+	test( 'should not duplicate existing metadata edits', () => {
```

Comment:
I like how the PostEditStore simplifies its list of edits so that it has as few edits as possible to accomplish the same result.  (I get that this is probably mostly to make the eventual network request feasible since the server doesn't want to hear about deleting and creating the same key)